kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -247,17 +237,83 @@ public class TableConfig<T> {
             }
           });
 
+  ///////////////////////////
+  // Static Helper Methods //
+  ///////////////////////////
+
+  /**
+   * Validates that the given properties have the delta prefix in the key name, and they are in the
+   * set of valid properties. The caller should get the validated configurations and store the case
+   * of the property name defined in TableConfig.
+   *
+   * @param configurations the properties to validate
+   * @throws InvalidConfigurationValueException if any of the properties are invalid
+   * @throws UnknownConfigurationException if any of the properties are unknown
+   */
+  public static Map<String, String> validateProperties(Map<String, String> configurations) {
+    Map<String, String> validatedConfigurations = new HashMap<>();
+    for (Map.Entry<String, String> kv : configurations.entrySet()) {
+      String key = kv.getKey().toLowerCase(Locale.ROOT);
+      String value = kv.getValue();
+      if (key.startsWith("delta.") && VALID_PROPERTIES.containsKey(key)) {
+        TableConfig<?> tableConfig = VALID_PROPERTIES.get(key);
+        if (tableConfig.editable) {
+          tableConfig.validate(value);
+          validatedConfigurations.put(tableConfig.getKey(), value);
+        } else {
+          throw DeltaErrors.cannotModifyTableProperty(kv.getKey());
+        }
+      } else {
+        throw DeltaErrors.unknownConfigurationException(kv.getKey());
+      }
+    }
+    return validatedConfigurations;
+  }
+
+  /**
+   * Parses the given JSON string into a map of key-value pairs.
+   *
+   * <p>The JSON string should be in the format:
+   *
+   * <pre>{@code {"key1": "value1", "key2": "value2", ...}}</pre>
+   *
+   * where both keys and values are strings.
+   *
+   * @param jsonString The JSON string to parse
+   * @return A map containing the key-value pairs extracted from the JSON string
+   */
+  public static Map<String, String> parseJSONKeyValueMap(String jsonString) {
+    if (jsonString == null || jsonString.trim().isEmpty()) {
+      return Collections.emptyMap();
+    }
+
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      return mapper.readValue(jsonString, new TypeReference<Map<String, String>>() {});
```

Comment:
no need to use the engine to do fancy row and columnar batch stuff. just use jackson